src/com/gr1t/taskmanager/TaskManager.java

```diff
@@ -1,19 +1,42 @@
 package com.gr1t.taskmanager;
 
 import java.util.*;
+import java.io.*;
 
 public class TaskManager {
-    static List<String> tasks = new ArrayList<>();
+    private static List<Task> tasks = new ArrayList<>();
 
     public static Task createTask(String title, String description, Priority priority) {
         Task task = new Task(title, description, priority);
-        tasks.add(String.valueOf(task));
+        tasks.add(task);
         return task;
 }
 
     public static void  printAllTasks() {
-        for (String task : tasks) {
-            System.out.println(task);
+        int index = 1;
+        for (Task task : tasks) {
+            System.out.println(index + " " + task);
+            index++;
         }
     }
+
+    public static void saveTasks (String fileName) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(fileName + ".txt"));) {
+            for (Task task : tasks) {
+                writer.write(String.valueOf(task));
+                writer.newLine();
+            }
+            System.out.println("File successfully saved as: " + fileName + ".txt");
+        } catch (IOException e) {
+            System.out.println("Your fault is: " + e.getMessage());
+        }
+    }
+
+    public static void sortByPriority() {
+        tasks.sort(Comparator.comparing(Task::getPriority));
+    }
+
+    public static void removeTask (int removeByIndex) {
+            tasks.remove(removeByIndex - 1);
```

Comment:
What happens if there's no index? thus the caller shouldn't keep track the inner indexes. Can be solved by either removing it by the  title/uuid